pkg/coinbase/validator.go

```diff
@@ -33,12 +89,11 @@ func (v Validator) ToString() string {
 	)
 }
 
-func (c *Client) ListValidators(ctx context.Context, networkId string, assetId string) ([]Validator, error) {
-	validatorList, _, err := c.client.ValidatorsAPI.ListValidators(
-		ctx,
-		normalizeNetwork(networkId),
-		assetId,
-	).Execute()
+func (c *Client) ListValidators(ctx context.Context, networkId string, assetId string, status *ValidatorStatus) ([]Validator, error) {
+	listValidatorReq := c.client.ValidatorsAPI.ListValidators(ctx, normalizeNetwork(networkId), assetId)
+	listValidatorReq = listValidatorReq.Status(getAPIValidatorStatus(status))
```

Comment:
I wanted to unit test the fact that we set an appropriate api status depending on the user facing status, but the `status` field in the `ApiListValidatorsRequest` from the generated code isn't exported so cannot do something like this

```
	s.mockValidatorsAPI.On("ListValidatorsExecute", mock.MatchedBy(func(req api.ApiListValidatorsRequest) bool {
		return req.status != nil && *req.status == api.VALIDATORSTATUS_ACTIVE
	})).Return(mockValidators, nil, nil)
```

open to any other ideas on how I can unit test this part.

Reply:
In the mock you can validate the parameters:

```
s.mockSuccessfulGetValidator(ctx, networkId, assetId, validatorId, &api.Validator{
		ValidatorId: validatorId,
		Status:      api.VALIDATORSTATUS_ACTIVE,
	})
```

Reply:
So I think you are doing that!

Reply:
The particular issue I listed is with `ListValidators` where I would like to assert the fact that if a user is filtering based on a certain status, then the request object I construct while making the backend api call, has the appropriate mapped status populated. That's where my above comment could have been useful if the status filed was. exported.

so yes, within `mockSuccessfulListValidators` I can validate but because the status filed is not exported I "think" my hands are tied.